### PR TITLE
Net core 2.0 implementation

### PR DESCRIPTION
**Pull request by qixu001**
Add dot net core 2.0 support
1) add dotnet core 2.0 projects for sdk, sample and test folder.
And aliyun-oss-sdk-dotnetcore.sln has the dot net core 2.0 projects only. 
while aliyun-oss-sdk.sln has all projects. 
2) Fix the test cases for dotnet core 2.0.
To work with dotnet core 2.0, the supported NUnit framework is NUnit 3.9 and it replaced the attributes TestFixtureTearDown  and TestFixtureTearSetup with OnetimeTearDown and OnetimeSetup.
3) Fix resource for dotnet core 2.0
Due to a dotnet core's bug (https://github.com/dotnet/coreclr/issues/12262, https://github.com/Microsoft/msbuild/issues/2221) in Linux, the MimeData could not load the content of the text file correctly. So to workaround this bug, just put the content in the resource file instead. 
4) other code fixes
4.1) KeyedHashAlgorithm.Create("SHA1")) does not work anymore in dotnet core 2.0. So create the HMACSHA1 instance directly.
4.2) In sdk/Common/Communication/ServiceClientImpl.cs, always use mono way to call the reflection code when it's dotnet core 2.0

### CODE REVIEW

File: sdk/Common/Communication/ServiceClientImpl.cs [hunk 1]
```diff
@@ -461,7 +461,13 @@ internal static class HttpExtensions
         internal static void AddInternal(WebHeaderCollection headers, string key, string value)
         {
             if (_isMonoPlatform == null)
+            {
+#if NETCOREAPP2_0
+                _isMonoPlatform = true;
```
Comment (baiyubin) [inline sdk/Common/Communication/ServiceClientImpl.cs]: -  `_isMonoPlatform` 是否运行在mono平台上，.net core 也可以在win平台上运行的，请确认下这个标志。

File: aliyun-oss-sdk-dotnetcore.sln [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
```
[the remark this answers is not in the document]
Comment (qixu001) [reply aliyun-oss-sdk-dotnetcore.sln]: 有两个sln主要是为了让那些只关心dotnetcore的人有一个独立的工程文件，这样编译的时候就不需要编译现在的mono版本了。AWS貌似也有很多给不同的dotnet版本的sln文件。

File: samples/aliyun-oss-sdk-samples-dotnetcore.csproj [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      <Visible>False</Visible>
+      <ProductName>.NET Framework 3.5 SP1 Client Profile</ProductName>
+      <Install>false</Install>
+    </BootstrapperPackage>
+    <BootstrapperPackage Include="Microsoft.Net.Framework.3.5.SP1">
+      <Visible>False</Visible>
+      <ProductName>.NET Framework 3.5 SP1</ProductName>
+      <Install>true</Install>
+    </BootstrapperPackage>
+    <BootstrapperPackage Include="Microsoft.Windows.Installer.3.1">
```
[the remark this answers is not in the document]
Comment (qixu001) [reply samples/aliyun-oss-sdk-samples-dotnetcore.csproj]: ok

File: sdk/Common/Communication/ServiceClientImpl.cs (hunk 1 above)
Comment (qixu001) [reply sdk/Common/Communication/ServiceClientImpl.cs]: 嗯，在Windows上运行过dotnet core，没问题。

**Pull request merged**